ibis/expr/types/relations.py

```diff
@@ -1129,6 +1129,42 @@ def materialize(self) -> TableExpr:
         return self
 
     def alias(self, alias: str) -> ir.TableExpr:
+        """Create a table expression with a specific name `alias`.
```

Comment:
Add a note/warning about the side effect?

Reply:
Yep, I'll make it clear that this side is in no way part of the public API.

Reply:
Done.